### PR TITLE
[EOS-20439] split setup provisioner command into granular one's

### PR DESCRIPTION
**Pull request by mazinamdar**
Signed-off-by: mazinamdar <mazhar.inamdar@seagate.com>

1. Bootstrap command copy of setup_provisioner with few responsiblility (salt , provisioner install and config)
2. Setup gluster will be called from bootstrap ( gluster configuration)
3. post_provisioner (cluster_id, machine_id , enclosure_id , encrypt pillar etc)

NOTE: For now I have used same parameters for all commands.

### CODE REVIEW

**Review by ypise — APPROVED** [review]

**Review by andkononykhin2 — COMMENTED** [review]
@mazinamdar Please check minor comments.

File: api/python/provisioner/commands/bootstrap_provisioner.py [hunk 1]
```diff
@@ -0,0 +1,1580 @@
...
+                        if not run_args.pypi_repo
+                        else 'https://pypi.org/'
+                    )
+                }
+            }
+            dump_yaml(pillar_path,  pillar)
+
+        return pillar
+
+    def _prepare_glusterfs_pillar(self, profile_paths, in_docker=False):
```
[the remark this answers is not in the document]
Comment (mazinamdar) [reply api/python/provisioner/commands/bootstrap_provisioner.py]: removed , not required. Comment is addressed

File: api/python/provisioner/commands/bootstrap_provisioner.py [hunk 2]
```diff
@@ -0,0 +1,1580 @@
...
+                f"{run_args.source} provisioner source is not supported yet"
+            )
+
+        # TODO EOS-18920: Wrap complete HA logic
+        # to a separate module in commands/bootstrap/
+
+        if run_args.ha:
+            SetupGluster(setup_ctx=setup_ctx,
+                         nodes=run_args.nodes).run(nodes, **kwargs)
+        # HA LOGIC END TODO wrap to a method / funciton
```
[the remark this answers is not in the document]
Comment (mazinamdar) [reply api/python/provisioner/commands/bootstrap_provisioner.py]: Comment is addressed

**Review by andkononykhin2 — APPROVED** [review]
@mazinamdar Approved. But I believe this new set of API calls shouldn't be exposed to users and used in production pipelines until sufficient testing is complete.
CC @ypise

**Comment by ypise** [comment]
> @mazinamdar Approved. But I believe this new set of API calls shouldn't be exposed to users and used in production pipelines until sufficient testing is complete.
> CC @ypise

The idea for this modularization is for internal flexibility.
Factory and Field setups would need different functionality to be called/executed. This is a preparatory step in that direction.

**Comment by ypise** [comment]
@mazinamdar I've updated a method to address Codacy issue, take a look at it:
https://github.com/Seagate/cortx-prvsnr/pull/1235/files#diff-c464509d9177edbc6ab5d22b37bb4469db55f02b0a12718a0622cf4a6210a221R304

**Pull request merged**